src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -95,7 +97,14 @@ private TupleTypeSymbol(ImmutableArray<Location> locations, NamedTypeSymbol unde
                 Emit.NoPia.EmbeddedTypesManager.IsValidEmbeddableType(underlyingType, syntax, diagnostics);
             }
 
-            return Create(underlyingType, elementNames, locationOpt, elementLocations);
+            var constructedType = Create(underlyingType, elementNames, locationOpt, elementLocations);
+
+            if (shouldCheckConstraints && (object)locationOpt != null)
```

Comment:
Why check `locationOpt` when we're passing `syntax` to `CheckConstraints`? Consider passing `locationOpt` to `CheckConstraints` instead.

Same comment for VB.